Timeout failures in `gazelle_generation_test` are difficult to understand.

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What version of gazelle are you using?

Post 0.28.0 commit: 498f2fc081afda84b82d444880bfedfa49c18baa

### What version of rules_go are you using?

0.36.0

### What version of Bazel are you using?

5.3.2

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

MacOS, x86_64

### What did you do?

Execute a `gazelle_binary` in a `gazelle_generation_test` that takes longer than the specified or default timeout.

### What did you expect to see?

An error telling me that the command timed out.

### What did you see instead?

```
--- FAIL: TestFullGeneration (3.41s)
    --- FAIL: TestFullGeneration/MySwiftPackage (3.41s)
        files.go:192: expected gazelle exit code: 0
            got: -1
```

--- a/testtools/files.go
+++ b/testtools/files.go
@@ -35,6 +35,8 @@ import (
 	"github.com/google/go-cmp/cmp"
 )
 
+const cmdTimeoutOrInterruptExitCode = -1
+
 // FileSpec specifies the content of a test file.
 type FileSpec struct {
 	// Path is a slash-separated path relative to the test directory. If Path
@@ -341,9 +343,14 @@ Run %s to update BUILD.out and expected{Stdout,Stderr,ExitCode}.txt files.
 		errs := make([]error, 0)
 		actualExitCode = cmd.ProcessState.ExitCode()
 		if config.ExitCode != actualExitCode {
-			errs = append(errs, fmt.Errorf("expected gazelle exit code: %d\ngot: %d",
-				config.ExitCode, actualExitCode,
-			))
+			if actualExitCode == cmdTimeoutOrInterruptExitCode {
+				errs = append(errs, fmt.Errorf("gazelle exceeded the timeout or was interrupted"))
+			} else {
+
+				errs = append(errs, fmt.Errorf("expected gazelle exit code: %d\ngot: %d",
+					config.ExitCode, actualExitCode,
+				))
+			}
 		}
 		actualStdout := redactWorkspacePath(stdout.String(), workspaceRoot)
 		if strings.TrimSpace(config.Stdout) != strings.TrimSpace(actualStdout) {

